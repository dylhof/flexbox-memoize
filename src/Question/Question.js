import React from 'react';

function Question(props) {
  const answers = props.currentQuestion.answers
  return(
    <div className='question-div'>
      <h2>{props.currentQuestion.property}</h2>
      <p>If you included this property in your CSS, 
        what image most closely resembles the result?</p>
      <div>
        {answers.map(answer => {
          return (
            <img
              className="answer--img"
              src={answer}
              alt={answer}
              key={answer}
              onClick={props.checkAnswer}
            />
          );
        })}
      </div>
    </div>
  )
}

export default Question;