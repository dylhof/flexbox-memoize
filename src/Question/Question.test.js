import React from 'react';
import Question from './Question';
import { shallow } from 'enzyme';

const question = {
  "property": "flex-direction: row",
  "correctAnswer": "https://i.imgur.com/Xo1MD86.png",
  "answers": [
  "https://i.imgur.com/Xo1MD86.png",
  "https://i.imgur.com/JIJHnVf.png",
  "https://i.imgur.com/4y9cZDw.png",
  "https://i.imgur.com/Id0kLlR.png"
  ],
  "info": "This establishes the main-axis, thus defining the direction flex items are placed in the flex container. Using row organizes the items from left to right."
  }
  const checkAnswerMock = jest.fn();
  const randomAnswers = [
    "https://i.imgur.com/Xo1MD86.png"
    ]

describe('Question', () => {
  let wrapper;
  beforeEach(() => {
    wrapper = shallow(
      <Question
        currentQuestion={question}
        checkAnswer={checkAnswerMock}
        randomAnswers={randomAnswers}/>
    )
  })

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  })

  it('should call check Answer when an answer is checked', () => {
    wrapper.find('.answer-img').simulate('click');
    expect(checkAnswerMock).toBeCalled();
  })

})