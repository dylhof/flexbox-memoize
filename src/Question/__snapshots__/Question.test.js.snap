// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Question should match the snapshot 1`] = `
<div
  className="question-div"
>
  <div
    className="question-title-div"
  >
    <h2>
      flex-direction: row
    </h2>
    <p>
      If you included this property in your CSS, what image most closely resembles the result?
    </p>
  </div>
  <div
    className="question-img-div"
  >
    <img
      alt="https://i.imgur.com/Xo1MD86.png"
      className="answer-img"
      key="https://i.imgur.com/Xo1MD86.png"
      onClick={[MockFunction]}
      src="https://i.imgur.com/Xo1MD86.png"
    />
  </div>
</div>
`;
